test/unit/parser/parser_unit_test.c

```diff
@@ -0,0 +1,440 @@
+
+		return -EINVAL;
+	}
+
+	l_free((char*)json_str);
+
+	return 0;
+}
+
+static int test_auth_json_create(void)
+{
+	char *json_str;
+	int err;
+
+	json_str = parser_auth_json_create("test_device_id",
+					   "test_device_token");
+
+	err = strcmp("{ \"id\": \"test_device_id\", \"token\":"
+		     " \"test_device_token\" }", json_str);
+
+	if(err != 0 ) {
+		l_free((char *) json_str);
+
+		return -EINVAL;
+	}
+
+	l_free((char *) json_str);
+
+	return 0;
+
+}
+
+static int test_unregister_json_create(void)
+{
+	char *json_str;
+	int err;
+
+	json_str = parser_unregister_json_create("test_device_id");
+
+	err = strcmp("{ \"id\": \"test_device_id\" }", json_str);
+
+	if(err != 0 ) {
+		l_free((char *) json_str);
+
+		return -EINVAL;
+	}
+
+	l_free((char *) json_str);
+
+	return 0;
+}
+
+static int test_get_key_str_from_json_str(void)
+{
+	char *value_str;
+	int err;
+
+	value_str = parser_get_key_str_from_json_str("{ \"id\": "
+					"\"test_device_id\", \"token\":"
+					" \"test_device_token\" }","id");
+
+	err = strcmp("test_device_id", value_str);
+
+	if(err != 0 ) {
+		l_free((char *) value_str);
+
+		return -EINVAL;
+	}
+
+	l_free((char *) value_str);
+
+	return 0;
+}
+
+static int test_is_key_str_or_null(void)
+{
+	bool res;
+
+	res = parser_is_key_str_or_null("{ \"TEST_KEY\": "
+					"\"test_device_id\", \"token\":"
+					" \"test_device_token\" }", "TEST_KEY");
+
+	if(!res) {
+
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int test_update_to_list(void)
+{
+	struct l_queue *queue;
+	knot_msg_data *msg;
+	char *json_str;
+
+	queue = parser_update_to_list("{\"id\": \"fbe64efa6c7f717e\",\"data\":"
+				      " [{\"sensorId\": 1,\"value\": false},"
+				      "{\"sensorId\": 2,\"value\": 1000}]}");
+
+	msg = l_queue_pop_head(queue);
+
+	if(msg->sensor_id != 1 || msg->payload.val_i != 0) {
+		goto error;
+	}
+	l_free(msg);
+
+	msg = l_queue_pop_head(queue);
+
+	if(msg->sensor_id != 2 || msg->payload.val_i != 1000) {
+		goto error;
+	}
+
+	l_queue_destroy(queue, l_free);
+	l_free(msg);
+
+	return 0;
+error:
+	l_queue_destroy(queue, l_free);
+	l_free(msg);
+
+	return -EINVAL;
+}
+
+static int test_request_to_list(void)
+{
+	struct l_queue *queue;
+	int *sensor_id;
+
+	queue = parser_request_to_list("{\"id\": \"fbe64efa6c7f717e\","
+				       "\"sensorIds\":[12,23,44]}");
+
+	sensor_id = l_queue_pop_head(queue);
+
+	if(sensor_id == NULL)
+		goto error;
+
+	if(*sensor_id != 12)
+		goto error;
+
+	l_free(sensor_id);
+	sensor_id = l_queue_pop_head(queue);
+
+	if(sensor_id == NULL)
+		goto error;
+
+	if(*sensor_id != 23)
+		goto error;
+
+	l_free(sensor_id);
+	sensor_id = l_queue_pop_head(queue);
+
+	if(sensor_id == NULL)
+		goto error;
+
+	if(*sensor_id != 44)
+		goto error;
+
+	l_queue_destroy(queue, l_free);
+	l_free(sensor_id);
+
+	return 0;
+error:
+	l_queue_destroy(queue, l_free);
+	l_free(sensor_id);
+
+	return -EINVAL;
+}
+
+static void *check_json_integrity(const char *id, const char *name,
+		struct l_queue *schema)
```

Comment:
Please be sure that the struct l_queue parameter is alligned with const char *id

Reply:
Done :rocket: